### PR TITLE
regex

### PR DESCRIPTION
**Pull request by ksdfg**
- add basic regex files
- use run_async parameter of Handler instead of wrapper in ud.py
- use sed via subprocess to replace text using regex
- return if command message doesn't have any reply to message

### CODE REVIEW

File: telebot/modules/regex.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    if "regex" in get_exceptions_for_chat(update.effective_chat.id):
+        return
+
+    if not update.effective_message.reply_to_message:
+        update.effective_message.reply_text("Gimme text to replace stuff in, baka!")
+        return
+
+    # make sure `s` command is properly terminated
+    command = update.effective_message.text
+    delimiter = command[1]
```
Comment (akhilnarang) [inline telebot/modules/regex.py:28]: Shouldn't we check what it is, exactly?

if command [1] in ('/',',''|') kinda

File: telebot/modules/regex.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    if command.count(delimiter) == 2:
```
Comment (akhilnarang) [inline telebot/modules/regex.py:29]: And also add a check for !=3, assuming you can safely escape other occurences not in the expression

File: telebot/modules/regex.py (hunk 1 above)
Comment (ksdfg) [reply telebot/modules/regex.py:28]: I mean, we're using a message handler with a regex filter to trigger the command anyway, so won't this check be a bit too redundant?

File: telebot/modules/regex.py (hunk 2 above)
Comment (ksdfg) [reply telebot/modules/regex.py:29]: can you elaborate a bit more?

File: telebot/modules/regex.py (hunk 1 above)
Comment (akhilnarang) [reply telebot/modules/regex.py:28]: Missed that. Nvm

File: telebot/modules/regex.py (hunk 2 above)
Comment (akhilnarang) [reply telebot/modules/regex.py:29]: Basically counting the number of delimiters, to predict an error beforehand itself

File: telebot/modules/regex.py (hunk 1 above)
Comment (akhilnarang) [reply telebot/modules/regex.py:28]: Missed that. Nvm

File: telebot/modules/regex.py (hunk 2 above)
Comment (ksdfg) [reply telebot/modules/regex.py:29]: but what would a !=3 check add to it? we need there to be 3 delimiters for it to be valid, and the regex will match one with minimum 2 delimiters, hence the check for 2 delimiters. Plus in the stage after this I remove everything except for flags after third delimiter, so cases like someone passing 4 delimiters etc. are also handled in that everything after the flags is discarded

Comment (akhilnarang) [reply telebot/modules/regex.py:29]: Hmm, basically I just wanted to try to detect errors in it pehle se, instead of sed returning errors

Comment (akhilnarang) [reply telebot/modules/regex.py:29]: Otherwise, seems fine

Comment (ksdfg) [reply telebot/modules/regex.py:29]: I thought sed returning errors would be the easiest way of getting proper error messages to user and tell them exactly what's wrong :laughing:

Comment (akhilnarang) [reply telebot/modules/regex.py:29]: In that way, yep, I was thinking of more possible vulnerabilities through that though xD

Comment (ksdfg) [reply telebot/modules/regex.py:29]: > Otherwise, seems fine

approve then, I wanna merge.....

**Review by akhilnarang — APPROVED** [review]

**Pull request merged**